### ISSUE TITLE
Extended percussion instruments missing from the matrix editor

### ISSUE THREAD

**Issue by buryoutougen** [helge17/tuxguitar#131]
There are some percussion sounds available below note 35 and above 81, but you can't input them using the matrix editor. You need to use an external reference like [this](https://en.wikipedia.org/wiki/General_MIDI#GS_extensions) or go by trial and error.

**Comment by guiv42** [comment]
Reading the wikipedia article you mentioned, I understand that only notes 35 to 81 are part of "GM standard MIDI". Outside of this range:

> Roland GS is a superset of the General MIDI standard that added several proprietary extensions

So I'm not sure it's a good idea to add these **proprietary** extensions, as users may export MIDI files from TuxGuitar to play them with other devices than Roland. Depending from the MIDI synthetizer used, I'm not even sure TuxGuitar would be able to play these sounds.
In other words, I would prefer to keep actual behavior unchanged.
Is my interpretation of this article correct?

**Comment by buryoutougen** [comment]
Sorry, maybe I wasn't completely clear, or I don't fully understand what the problem is.

These sounds are already in TuxGuitar, it's just that you have to input the note numbers manually, as they can't be selected from the matrix editor.

I'm not an expert on MIDI either, but my understanding is that these additional notes were added by Roland and then standardized in General MIDI 2 (also see [here](https://soundprogramming.net/file-formats/general-midi-level-2/) and [here](https://soundprogramming.net/file-formats/general-midi-drum-note-numbers/)).

Anyway, it's not a big deal, I just thought it's something that could be improved, as someone might not be aware of these additional sounds if they only use the matrix editor to input drums.

**Comment by guiv42** [comment]
OK, got it, thanks for your explanation.
This should be easy to add, pull request coming soon.
Note: all these sound names are hard-coded in English, that might be interesting to open them for translation one day...

**Comment by guiv42** [comment]
Is it just my configuration, or is there a problem of sound within the matrix editor?
When I click in one cell of the matrix, it produces the expected sound. If, then, I click in another cell, the sound I hear corresponds to my first click, not the second instrument... Clicking twice in the second cell (remove / add sound again) and it produces the correct sound.

**Comment by buryoutougen** [comment]
Yes, same for me. It seems to me that when you click on a cell, TuxGuitar first plays the notes from the currently focused column, then switches focus to the column you clicked - should be the other way around.

**Comment by guiv42** [comment]
OK, I understood the cause of that problem, it should be fixed by pull request #134.
I will keep you informed when the modifications are available in a pre-release, so you can test them, and close this issue if you think it's OK. Stay tuned!

**Comment by buryoutougen** [comment]
I tested the new pre-release, it works correctly. However, I've noticed there are 3 more sounds outside of GM2 available in TuxGuitar - notes 24, 25 and 26, but I'm not sure what instruments they represent and how they should be named (26 sounds like another variation of snare, 24 and 25 like different-sized versions of some kind of low drum).

If you think these sounds should be ignored in the matrix editor, I will close this issue.

**Comment by guiv42** [comment]
Thanks for your feedback.
Since I can find no description of these notes 24 to 26 neither in GM1, GM2, nor in the different links above, it seems reasonable to me to ignore them in the matrix editor
